aws/resource_aws_codebuild_project.go

```diff
@@ -451,6 +465,7 @@ func resourceAwsCodeBuildProjectRead(d *schema.ResourceData, meta interface{}) e
 	d.Set("name", project.Name)
 	d.Set("service_role", project.ServiceRole)
 	d.Set("build_timeout", project.TimeoutInMinutes)
+	d.Set("badge_url", project.Badge.BadgeRequestUrl)
```

Comment:
In order to prevent potential nil dereference panic, we need to wrap these in a nil check:

```go
if project.Badge != nil {
  d.Set("badge_enabled", project.Badge.BadgeEnabled)
  d.Set("badge_url", project.Badge.BadgeRequestUrl)
} else {
  d.Set("badge_enabled", false)
  d.Set("badge_url", "")
}
```